app/src/main/java/org/oppia/app/player/audio/LanguageDialogFragment.kt

```diff
@@ -6,75 +6,55 @@ import android.os.Bundle
 import androidx.appcompat.app.AlertDialog
 import androidx.fragment.app.DialogFragment
 import org.oppia.app.R
-import java.util.ArrayList
 
 private const val KEY_LANGUAGE_LIST = "LANGUAGE_LIST"
-private const val KEY_CURRENT_LANGUAGE = "CURRENT_LANGUAGE"
+private const val KEY_SELECTED_INDEX = "SELECTED_INDEX"
 
 /**
- * DialogFragment that controls language selection in audio and written translations
+ * DialogFragment that controls language selection in audio and written translations.
  */
 class LanguageDialogFragment : DialogFragment() {
   companion object {
-    lateinit var languageInterface1: LanguageInterface
     /**
-     * This function is responsible for displaying content in DialogFragment
-     * @param languageInterface: [LanguageInterface] to send data back to parent
-     * @param languageCodeList: List of strings containing languages
+     * This function is responsible for displaying content in DialogFragment.
+     *
+     * @param languageArrayList: List of strings containing languages
      * @param currentLanguageCode: Currently selected language code
-     * @return LanguageDialogFragment: DialogFragment
+     * @return [LanguageDialogFragment]: DialogFragment
      */
     fun newInstance(
-      languageInterface: LanguageInterface,
-      languageCodeList: List<String>,
+      languageArrayList: ArrayList<String>,
       currentLanguageCode: String
     ): LanguageDialogFragment {
-
-      languageInterface1 = languageInterface
-      val selectedIndex: Int = languageCodeList.indexOf(currentLanguageCode)
-      val fragment = LanguageDialogFragment()
+      val selectedIndex = languageArrayList.indexOf(currentLanguageCode)
+      val languageDialogFragment = LanguageDialogFragment()
       val args = Bundle()
-      args.putStringArrayList(
-        KEY_LANGUAGE_LIST,
-        languageCodeList as ArrayList<String>
-      )
-      args.putInt(KEY_CURRENT_LANGUAGE, selectedIndex)
-      fragment.arguments = args
-      return fragment
+      args.putStringArrayList(KEY_LANGUAGE_LIST, languageArrayList)
+      args.putInt(KEY_SELECTED_INDEX, selectedIndex)
+      languageDialogFragment.arguments = args
+      return languageDialogFragment
     }
   }
 
   override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-    var languageList = savedInstanceState?.getStringArrayList(KEY_LANGUAGE_LIST)
-    var currentIndex = savedInstanceState?.getInt(KEY_CURRENT_LANGUAGE, 0)
-
-    if (languageList != null) {
-      // Not null
-      // Currently data is not getting transferred to LanguageDialogFragment.
-    } else {
-      // Null
-      // Error handling can be done here if needed.
-      languageList = ArrayList<String>()
-      languageList.add("en")
-      languageList.add("hi")
-      languageList.add("hi-en")
-    }
+    var selectedIndex = arguments!!.getInt(KEY_SELECTED_INDEX, 0)
+    val languageArrayList: ArrayList<String> = arguments?.getStringArrayList(KEY_LANGUAGE_LIST) as ArrayList<String>
```

Comment:
Why is this cast needed? It seems superfluous since getStringArrayList() returns ArrayList<String>.

Also, the ``?.`` operator needed here? I think you can avoid it by saving arguments locally:

```kotlin
val args = checkNotNull(arguments) { "Expected arguments to be pass to LanguageDialogFragment" }
// args can now be used without the nullability considerations
```

Reply:
Regarding extra cast:

`val languageArrayList: ArrayList<String> = args.getStringArrayList(KEY_LANGUAGE_LIST) as ArrayList<String>`

If we do not add this extra cast then there is type mismatch warning
<img width="509" alt="Screenshot 2019-09-25 at 11 18 20 AM" src="https://user-images.githubusercontent.com/9396084/65572607-376cce00-df86-11e9-9f2f-8c9516e33ba5.png">